source/mcpele/uniform_spherical_sampling.h

```diff
@@ -0,0 +1,36 @@
+#ifndef _MCPELE_UNIFORM_SPHERICAL_SAMPLING_H__
+#define _MCPELE_UNIFORM_SPHERICAL_SAMPLING_H__
+
+namespace mcpele {
+
+/**
+ * Sample points uniformly at random within an N-ball.
+ * See also: http://mathworld.wolfram.com/BallPointPicking.html
+ */    
+
+class UniformSphericalSampling : public TakeStep {
+protected:
+    std::mt19937_64 m_gen;
+    const double m_radius2;    
+    std::uniform_real_distribution<double> m_dist;
+public:
+    virtual ~UniformSphericalSampling() {}
+    UniformSphericalSampling(const size_t seed=42, const double radius=1)
+        : m_gen(seed),
+          m_radius2(radius * radius),
+          m_dist(-radius, radius)
+    {}
+    void set_generator_seed(const size_t inp) { m_gen.seed(inp); }
+    virtual void displace(pele::Array<double>& coords, MC* mc)
+    {
+        do {
+            for (size_t i = 0; i < coords.size(); ++i) {
+                coords[i] = m_dist(m_gen);
+            }
+        } while (pele::dot(coords, coords) > m_radius2);
```

Comment:
This type of rejection sampling will be exponentially slow for large dimensions when the volume of the sphere becomes minuscule compared to the volume of a cube.  I would guess that even for 3-dimensions it is very slower.  You can do this exactly, and without rejections, by first sampling a unit vector in n-dimensions, then sampling the radius from a power law distribution.  I can look up the exact details if you need them.  @smcantab and I used this a lot to do the nested sampling stuff.